tapchannel/aux_leaf_signer.go

```diff
@@ -240,36 +240,53 @@ func (s *AuxLeafSigner) processAuxSigBatch(chanState *channeldb.OpenChannel,
 
 	defer s.Wg.Done()
 
+	shutdownErr := fmt.Errorf("tapd is shutting down")
 	log.Tracef("Processing %d aux sig jobs", len(sigJobs))
 
 	for idx := range sigJobs {
 		sigJob := sigJobs[idx]
 		cancelAndErr := func(err error) {
 			log.Errorf("Error processing aux sig job: %v", err)
 
-			close(sigJob.Cancel)
+			// Check that the cancel signal was not already sent
```

Comment:
Since we're expected to be the only ones closing the `sigJob.Cancel` channel, we could also do this with a `sync.Once`. But I guess this works too and would also work in case someone else closes the channel.

Reply:
Yeah I think we need to straighten out exactly who/what can close the quit channel, otherwise we'll invariably run into a double close panic.

Reply:
lnd can close the channel, it is shared amongst the 'classic' sig jobs and these aux sig jobs.

Reply:
Updated s.t. lnd is the _only_ one that will be closing the channel (simpler flow, having tapd close was not really needed)